borg/remote.py

```diff
@@ -113,10 +113,13 @@ def serve(self):
                         # These exceptions are reconstructed on the client end in RemoteRepository.call_many(),
                         # and will be handled just like locally raised exceptions. Suppress the remote traceback
                         # for these, except ErrorWithTraceback, which should always display a traceback.
-                        if not isinstance(e, (Repository.DoesNotExist, Repository.AlreadyExists, PathNotAllowed)):
+                        if isinstance(e, (Repository.DoesNotExist, Repository.AlreadyExists, Repository.ObjectNotFound,
+                                          Repository.CheckNeeded, IntegrityError, PathNotAllowed, InvalidRPCMethod)):
+                            exc = e.args
+                        else:
+                            exc = "Remote Exception (see remote log for the traceback)"
                             logging.exception('Borg %s: exception in RPC call:', __version__)
                             logging.error(sysinfo())
-                        exc = "Remote Exception (see remote log for the traceback)"
```

Comment:
so, old borg serve code always gave a str for exc (on the remote side received as bytes due to msgpack).